### PR TITLE
Implemented pysam skipability

### PR DESCRIPTION
**Pull request by bocajnotnef**
issue: #1120
starting PR comment: https://github.com/dib-lab/khmer/pull/1019#discussion_r30709115

added stuff within the import tests to skip the test (or really just prevent it from failing) if the pysam module is not installed.

@ctb

### CODE REVIEW

**Comment by bocajnotnef** [comment]
relevant to #750

**Comment by ctb** [comment]
Ahh, apologies, I didn't realize that it was a pysam import error in the test_import_all test.  But...

This is an ugly and rather overbroad hack :). We don't want to check for failed pysam import in *generic* test running code!

Instead, I suggest moving 'import pysam' into the 'main()' function, and if/when we write an actual exec test for the script, we can make that test skip when pysam import fails.

**Comment by bocajnotnef** [comment]
Agreed, it is an ugly hack. 

I'll go move the things.

**Comment by bocajnotnef** [comment]
An embarassing number of commits later, tests are passing. 

@ctb Better?

File: sandbox/readaligner_pairhmm_train.py [hunk 1]
```diff
@@ -28,6 +27,11 @@ def trusted_str(cov, trusted_cutoff):
 
 
 def main():
+    try:
```
Comment (ctb) [inline sandbox/readaligner_pairhmm_train.py]: why try/pass here? just import it. tests should still succeed as we don't actually run this script in the tests.

Comment (bocajnotnef) [inline sandbox/readaligner_pairhmm_train.py]: I swear I'd dropped the try/except... Gah.

Comment (bocajnotnef) [inline sandbox/readaligner_pairhmm_train.py]: Aha, I see why I had that.

If the import fails the exec test fails:

```
test_sandbox_scripts.py: test import readaligner_pairhmm_train.py ... FAIL

======================================================================
FAIL: test_sandbox_scripts.py: test import readaligner_pairhmm_train.py
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/bocajnotnef/git/env/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/bocajnotnef/git/khmer/tests/test_sandbox_scripts.py", line 83, in __call__
    "{0}".format(traceback))
AssertionError: (u"/home/bocajnotnef/git/khmer/sandbox/readaligner_pairhmm_train.py cannot be exec'd", u"<module 'traceback' from '/usr/lib/python2.7/traceback.pyc'>")
```

Comment (ctb) [inline sandbox/readaligner_pairhmm_train.py]: OK, this is a bit of a hack, but if you move the import down below parse_args, you'll be fine.

**Comment by bocajnotnef** [comment]
@ctb Updated

File: sandbox/readaligner_pairhmm_train.py [hunk 2]
```diff
@@ -28,6 +27,7 @@ def trusted_str(cov, trusted_cutoff):
 
 
 def main():
+
```
Comment (ctb) [inline sandbox/readaligner_pairhmm_train.py]: extraneous blank line -> pls remove

**Comment by ctb** [comment]
thx!

File: sandbox/readaligner_pairhmm_train.py [hunk 3]
```diff
@@ -37,6 +36,8 @@ def main():
 
     args = parser.parse_args()
 
+    import pysamFOO
```
Comment (mr-c) [inline sandbox/readaligner_pairhmm_train.py]: ??? No, this does not work. Why was this merged?

Comment (ctb) [inline sandbox/readaligner_pairhmm_train.py]: Hah, I guess that's the problem with allowing it to be skipped!

Comment (mr-c) [inline sandbox/readaligner_pairhmm_train.py]: I thought we found a Python only sam parser? All I can come up with is
https://pypi.python.org/pypi/libsam/ which doesn't fill me with confidence

On Tue, Jul 28, 2015 at 11:00 AM C. Titus Brown <notifications@github.com>
wrote:

> In sandbox/readaligner_pairhmm_train.py
> <https://github.com/dib-lab/khmer/pull/1121#discussion_r35678960>:
>
> > @@ -37,6 +36,8 @@ def main():
> >
> >      args = parser.parse_args()
> >
> > +    import pysamFOO
>
> Hah, I guess that's the problem with allowing it to be skipped!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dib-lab/khmer/pull/1121/files#r35678960>.
>
-- 
Michael R. Crusoe: Programmer & Bioinformatician crusoe@ucdavis.edu
The lab for Data Intensive Biology; University of California, Davis
https://impactstory.org/MichaelRCrusoe http://twitter.com/biocrusoe

Comment (ctb) [inline sandbox/readaligner_pairhmm_train.py]: From https://twitter.com/ctitusbrown/status/626093435817193472,

https://github.com/mdshw5/simplesam
https://github.com/brentp/hts-python/

**Pull request merged**